server/v2/core/mempool/mempool.go

```diff
@@ -2,22 +2,43 @@ package mempool
 
 import (
 	"context"
+	"errors"
 
 	"cosmossdk.io/core/transaction"
 )
 
 // Mempool defines the required methods of an application's mempool.
 type Mempool[T transaction.Tx] interface {
 	// Insert attempts to insert a Tx into the app-side mempool returning
-	// an error upon failure. Insert will validate the transaction using the txValidator
-	Insert(ctx context.Context, txs T) error
+	// an error upon failure.
+	Insert(context.Context, T) error
 
-	// Get returns a list of transactions to add in a block
-	// where num is the number of txs to get. NOTE: size
-	// represents the size of a TX in bytes.
-	Get(ctx context.Context, size int) ([]T, error)
+	// Select returns an Iterator over the app-side mempool. If txs are specified,
+	// then they shall be incorporated into the Iterator. The Iterator must be
+	// closed by the caller.
+	Select(context.Context, []T) Iterator[T]
+
+	// CountTx returns the number of transactions currently in the mempool.
+	CountTx() int
```

Comment:
is this needed?

Reply:
It was part of the rollback, I'm not using it so I'll just remove it